src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -0,0 +1,903 @@
+					height = width / aspectRatio;
+					imageSize = new CGSize(width, height);
+					break;
+				default: // Fill
+					imageSize = Bounds.Size;
+					break;
+			}
+
+			// The ImageBrush.AlignementX/Y will tell us the LOCATION we need for the layer
+			double deltaX;
+			switch (imageBrush.AlignmentX)
+			{
+				case AlignmentX.Left:
+					deltaX = 0;
+					break;
+				case AlignmentX.Right:
+					deltaX = (double)(Bounds.Width - imageSize.Width);
+					break;
+				default: // Center
+					deltaX = (double)(Bounds.Width - imageSize.Width) * 0.5f;
+					break;
+			}
+
+			double deltaY;
+			switch (imageBrush.AlignmentY)
+			{
+				case AlignmentY.Top:
+					deltaY = 0;
+					break;
+				case AlignmentY.Bottom:
+					deltaY = (double)(Bounds.Height - imageSize.Height);
+					break;
+				default: // Center
+					deltaY = (double)(Bounds.Height - imageSize.Height) * 0.5f;
+					break;
+			}
+
+			var imageFrame = new CGRect(new CGPoint(deltaX, deltaY), imageSize);
+
+			// This is the layer with the actual image in it. Its frame is the inside of the border.
+			var imageLayer = new CALayer
+			{
+				Contents = uiImage.CGImage,
+				Frame = imageFrame,
+				MasksToBounds = true,
+			};
+
+			imageContainerLayer.AddSublayer(imageLayer);
+
+			return true;
+		}
+		#endregion
+
+		#region Helper methods
+		/// <summary>
+		/// Gets the rounded/adjusted half stroke thickness that should be used for measuring absolute shapes (Path, Line, Polyline and Polygon)
+		/// </summary>
+		private double GetHalfStrokeThickness()
+			=> Math.Floor((ActualStrokeThickness + .5) / 2.0);
+
+		private
+			(
+				double width, bool hasWidth, double height, bool hasHeight,
+				(double width, bool hasWidth, double height, bool hasHeight) min,
+				(double width, bool hasWidth, double height, bool hasHeight) max
+			)
+			GetUserSizes()
+		{
+			var width = Width;
+			var height = Height;
+			var minWidth = MinWidth;
+			var minHeight = MinHeight;
+			var maxWidth = MaxWidth.AtLeast(minWidth); // UWP is applying "min" after "max", so if "min" > "max", "min" wins
+			var maxHeight = MaxHeight.AtLeast(minHeight);
+
+			return (
+				width, !IsNaN(width), height, !IsNaN(height),
+				(minWidth, IsFinite(minWidth) && minWidth > 0, minHeight, IsFinite(minHeight) && minHeight > 0),
+				(maxWidth, IsFinite(maxWidth), maxHeight, IsFinite(maxHeight)));
+		}
+
+		// This replicates the behavior of LayoutHelper.GetMinMax() without reading again all DependencyProperties
+		private (Size min, Size max) GetMinMax(
+			(
+				double width, bool hasWidth, double height, bool hasHeight,
+				(double width, bool hasWidth, double height, bool hasHeight) min,
+				(double width, bool hasWidth, double height, bool hasHeight) max
+			)
+			userSize)
+		{
+			var size = new Size(userSize.width, userSize.height);
+			var minSize = new Size(userSize.min.width, userSize.min.height);;
+			var maxSize = new Size(userSize.max.width, userSize.max.height); ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5619334)